src/Campaigns/Routes/RegisterCampaignRoutes.php

```diff
@@ -0,0 +1,305 @@
+                    'permission_callback' => '__return_true',
+                ],
+                'args' => [
+                    'page' => [
+                        'type' => 'integer',
+                        'default' => 1,
+                        'minimum' => 1,
+                    ],
+                    'per_page' => [
+                        'type' => 'integer',
+                        'default' => 30,
+                        'minimum' => 1,
+                        'maximum' => 100,
+                    ],
+                ],
+            ]
+        );
+    }
+
+    /**
+     * Update Campaign route
+     *
+     * @unreleased
+     */
+    public function registerUpdateCampaign()
+    {
+        register_rest_route(
+            CampaignRoute::NAMESPACE,
+            CampaignRoute::CAMPAIGN,
+            [
+                [
+                    'methods' => WP_REST_Server::EDITABLE,
+                    'callback' => function (WP_REST_Request $request) {
+                        return $this->campaignRequestController->updateCampaign($request);
+                    },
+                    'permission_callback' => function () {
+                        return current_user_can('manage_options');
+                    },
+                ],
+                'args' => [
+                    'id' => [
+                        'type' => 'integer',
+                        'required' => true,
+                    ],
+                    'title' => [
+                        'type' => 'string',
+                        'sanitize_callback' => 'rest_sanitize_request_arg',
+                        'validate_callback' => 'rest_validate_request_arg'
+                    ],
+                ],
+                'schema' => [$this, 'getSchema'],
+            ]
+        );
+    }
+
+
+    /**
+     * Create Campaign route
+     *
+     * @unreleased
+     */
+    public function registerCreateCampaign()
+    {
+        register_rest_route(
+            CampaignRoute::NAMESPACE,
+            CampaignRoute::CAMPAIGNS,
+            [
+                [
+                    'methods' => WP_REST_Server::CREATABLE,
+                    'callback' => function (WP_REST_Request $request) {
+                        return $this->campaignRequestController->createCampaign($request);
+                    },
+                    'permission_callback' => function () {
+                        return current_user_can('manage_options');
+                    },
+                ],
+                'args' => [
+                    'title' => [
+                        'type' => 'string',
+                        'required' => true,
+                        'sanitize_callback' => 'sanitize_text_field',
+                    ],
+                    'description' => [
+                        'type' => 'string',
+                        'required' => false,
+                        'sanitize_callback' => 'sanitize_text_field',
+                    ],
```

Comment:
Let's add here these args:

```
 'image' => [
    'type' => 'string',
    'required' => false,
    'sanitize_callback' => 'sanitize_url',
],
'goalType' => [
    'type' => 'string',
    'required' => true,
    'sanitize_callback' => 'sanitize_text_field',
],
'goal' => [
    'type' => 'integer',
    'required' => true,
    'sanitize_callback' => 'absint',
],
```

This way, we can keep the changes that I did on `CreateCampaign.php` file: https://github.com/impress-org/givewp/pull/7543/files#diff-2adc6a525044d3090b82717de88d883fd6156a30099c8c238698db2f421900a3

Reply:
The schema should handle this automatically, and I don't want to duplicate code unnecessarily. Currently, the schema doesn't work for some reason. I'm working on it to figure out why.